### PR TITLE
Simplify parquet filter predicate test, fix Page Filtering Incorrectly Handles Pages with Different Row Counts 

### LINKED ISSUE #4744: Page Filtering Incorrectly Handles Pages with Different Row Counts
**Describe the bug**

`prune_pages_in_one_row_group` uses the size of the previous page when computing the next value of `sum_row`. This causes it to produce incorrect `RowSelection` when computing for a row group containing different size pages.

**To Reproduce**

**Expected behavior**

**Additional context**

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #4744

# Rationale for this change

# What changes are included in this PR?

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/core/tests/parquet/filter_pushdown.rs [hunk 1]
```diff
@@ -42,60 +42,64 @@ use tempfile::TempDir;
...
+    // enable logging so RUST_LOG works
+    let _ = env_logger::try_init();
+}
+
+#[cfg(not(target_family = "windows"))]
+#[tokio::test]
+async fn single_file() {
+    // Only create the parquet file once as it is fairly large
 
-    let mut set = tokio::task::JoinSet::new();
```
Comment (tustvold) [inline datafusion/core/tests/parquet/filter_pushdown.rs:80]: I reverted the JoinSet change as the test now runs in under 2 seconds, and it made debugging test failures difficult as the stack trace pointed to where the JoinHandle is unwrapped, not the case that actually failed. It also made the failures non-deterministic, which was also annoying

File: datafusion/core/tests/parquet/filter_pushdown.rs [hunk 2]
```diff
@@ -223,107 +226,123 @@ async fn single_file() {
...
-        // predicate is chosen carefully to prune pages 1, 2, 4, and 5
-        // time > 1970-01-01T00:00:00.004300000
-        .with_filter(col("time").gt(lit_timestamp_nano(4300000)))
+        // predicate is chosen carefully to prune all bar 0-1, 1-0, 1-1
+        // time > 1970-01-01T00:00:00.000216064
+        .with_filter(col("time").gt(lit_timestamp_nano(000216064)))
         .with_pushdown_expected(PushdownExpected::Some)
         .with_page_index_filtering_expected(PageIndexFilteringExpected::Some)
-        .with_expected_rows(9745)
+        .with_expected_rows(178)
```
Comment (tustvold) [inline datafusion/core/tests/parquet/filter_pushdown.rs:313]: This is currently failing, the correct answer is 178, for some reason we return less

File: datafusion/core/src/physical_plan/file_format/parquet/page_filter.rs [hunk 3]
```diff
@@ -251,7 +251,7 @@ fn prune_pages_in_one_row_group(
                 let mut sum_row = *row_vec.first().unwrap();
                 let mut selected = *values.first().unwrap();
                 trace!("Pruned to to {:?} using {:?}", values, pruning_stats);
-                for (i, &f) in values.iter().skip(1).enumerate() {
+                for (i, &f) in values.iter().enumerate().skip(1) {
```
Comment (tustvold) [inline datafusion/core/src/physical_plan/file_format/parquet/page_filter.rs:254]: This is the cause of #4744

**Review by alamb — APPROVED** [review]
Thank you @tustvold

File: datafusion/core/tests/parquet/filter_pushdown.rs (hunk 1 above)
Comment (waynexia) [reply datafusion/core/tests/parquet/filter_pushdown.rs:80]: Looks good to me. It also takes me some time to figure out which sub-case fails. If the time is greatly shortened, I agree we need not pay for the parallel

**Review by Ted-Jiang — APPROVED** [review]
Thanks for this 👍
Suffered covid these days , sorry for the late respond.

**Comment by tustvold** [comment]
I'm looking into the other test failures, I'm hoping the tests are wrong

File: datafusion/core/tests/parquet/page_pruning.rs [hunk 4]
```diff
@@ -108,14 +108,16 @@ async fn page_index_filter_one_col() {
 
     let batch = results.next().await.unwrap().unwrap();
 
-    // `month = 12` from the page index should create below RowSelection
-    //  vec.push(RowSelector::skip(894));
-    //  vec.push(RowSelector::select(339));
-    //  vec.push(RowSelector::skip(3330));
+    // `month = 1` or `month = 2` from the page index should create below RowSelection
```
Comment (tustvold) [inline datafusion/core/tests/parquet/page_pruning.rs:111]: These tests were just wrong, I have fixed them using https://github.com/apache/arrow-rs/pull/3405 to verify that they are correct

**Pull request merged**